From 1e3801d0cef748e599e79b013cdac0825bf13c5c Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Wed, 16 Nov 2016 18:56:39 +0200
Subject: [PATCH 172/278] drivers/tty/serial: Updated Baud Rate Calcualtion

The baud rate calculation was truncating instead of rounding. This
caused, in some cases, the initial baud rate to be set incorrectly.

Updated to use DIV_ROUND_CLOSEST().

Signed-off-by: John Jacques <john.jacques@intel.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 drivers/tty/serial/amba-pl011.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/amba-pl011.c b/drivers/tty/serial/amba-pl011.c
index 6b6fd82..70ca2ff 100644
--- a/drivers/tty/serial/amba-pl011.c
+++ b/drivers/tty/serial/amba-pl011.c
@@ -2295,7 +2295,8 @@ static void pl011_console_putchar(struct uart_port *port, int ch)
 		ibrd = pl011_read(uap, REG_IBRD);
 		fbrd = pl011_read(uap, REG_FBRD);
 
-		*baud = uap->port.uartclk * 4 / (64 * ibrd + fbrd);
+		*baud =	DIV_ROUND_CLOSEST(uap->port.uartclk * 4,
+					  64 * ibrd + fbrd);
 
 		if (uap->vendor->oversampling) {
 			if (pl011_read(uap, REG_CR)
-- 
1.8.5.2.233.g932f7e4

